source/_integrations/homeassistant.markdown

```diff
@@ -3,6 +3,116 @@ title: "Core integration"
 description: "Description of the homeassistant integration."
 logo: home-assistant.png
 ha_release: 0.0
+ha_qa_scale: internal
 ---
 
 The Home Assistant integration provides generic implementations like the generic `homeassistant.turn_on`.
+
+### Services
+
+the `homeassistant` integration provides services for controlling Home Assistant itself, as well as generic controls for any entity
+
+#### Service `homeassistant.check_config`
+
+Reads the config files and checks them for correctness, but **does not** load them into home assistant. Creates a persistant notification and log entry if errors are found.
+
+#### Service `homeassistant.check_config`
+
+Loads the main config file (`configuration.yaml`) and all linked files. Once loaded the new configuration is applied.
+
+#### Service `homeassistant.restart`
+
+Restarts the Home Assistant instance (also reloading the configuration on start).
+
+#### Service `homeassistant.stop`
+
+Stops the Home Assistant instance. Home Assistant must be restarted from the Host device to run again.
+
+#### Service `homeassistant.set_location`
+
+Update the location of the Home Assistant default zone (usually "Home").
+
+| Service data attribute    | Optional | Description                                           |
+|---------------------------|----------|-------------------------------------------------------|
+| `latitude`                |       no | Latitude of your location.                            |
+| `longitude`               |       no | Longitude of your location.                           |
+
+##### Example
+
+```yaml
+action:
+  service: homeassistant.set_location
+  data:
+    latitude: 32.87336
+    longitude: 117.22743
+```
+
+#### Service `homeassistant.toggle` 
+
+Generic service to toggle devices on/off under any domain. Same usage as the light.turn_on, switch.turn_on, etc. services.
+
+| Service data attribute    | Optional | Description                                           |
+|---------------------------|----------|-------------------------------------------------------|
+| `entity_id`               |       no | The entity_id of the device to toggle on/off.         |
```

Comment:
```suggestion
| `entity_id`               |       yes | The entity_id of the device to toggle on/off.         |
```

The entity ID is optional.

Reply:
the entity is not optional. See [code](https://github.com/home-assistant/home-assistant/blob/master/homeassistant/components/homeassistant/__init__.py#L46)

Reply:
A) It is optional in the service call schema.
B) It is optional, as entity id's can be a result of the area called, see:

https://github.com/home-assistant/home-assistant/blob/79b47157b85d70a87f89ac6317b2455bc4ff0860/homeassistant/helpers/service.py#L138

C) Don't resolve comments that aren't resolved by changing code, since I had to expand them all again to find this response.

Reply:
thanks for the explanation. sorry about resolving it!